src/Uno.UI/Controls/CommandBarHelper.iOS.cs

```diff
@@ -101,6 +101,15 @@ public static void PageDidDisappear(UIViewController pageController)
 			}
 		}
 
+		public static void PageWillDisappear(UIViewController pageController)
+		{
+			if (pageController.FindTopCommandBar() is { } topCommandBar)
+			{
+				// Set the native navigation bar to null so it does not render when the page is not visible
+				SetNavigationBar(topCommandBar, null!);
```

Comment:
`null!` is not a thing you should ever write if you can avoid it. Rather you should change the called signature to `SetNavigationItem(CommandBar commandBar, UIKit.UINavigationItem? navigationItem)`

Reply:
> `null!` is not a thing you should ever write if you can avoid it. Rather you should change the called signature to `SetNavigationItem(CommandBar commandBar, UIKit.UINavigationItem? navigationItem)`

 I just followed the implementation of similar methods ex: `PageDidDisappear` as I did not want to do refactoring on code I did not add.

@davidjohnoliver

Reply:
Will push another PR doing this refactoring